packages/amplify-codegen/src/utils/GraphQLStatementsFormatter.js

```diff
@@ -36,27 +51,42 @@ class GraphQLStatementsFormatter {
   }
 
   formatGraphQL(statements) {
-    const headerBuffer = this.headerComments.map( comment => `# ${comment}`).join(LINE_DELIMITOR);
+    const headerBuffer = this.headerComments.map(comment => `# ${comment}`).join(LINE_DELIMITOR);
     const statementsBuffer = statements ? [...statements.values()].join(LINE_DELIMITOR) : '';
     const formattedOutput = [headerBuffer, LINE_DELIMITOR, statementsBuffer].join(LINE_DELIMITOR);
     return formattedOutput;
   }
 
   formatJS(statements) {
     const lintOverridesBuffer = this.lintOverrides.join(LINE_DELIMITOR);
-    const headerBuffer = this.headerComments.map( comment => `// ${comment}`).join(LINE_DELIMITOR);
+    const headerBuffer = this.headerComments.map(comment => `// ${comment}`).join(LINE_DELIMITOR);
     const formattedStatements = [];
     if (statements) {
       for (const [key, value] of statements) {
-        formattedStatements.push(
-          `export const ${key} = /* GraphQL */ \`${value}\``
-        );
+        const typeTag = this.buildTypeTag(key);
+        formattedStatements.push(`export const ${key} = /* GraphQL */ \`${value}\`${typeTag}`);
       }
     }
-    const formattedOutput = [lintOverridesBuffer, headerBuffer, LINE_DELIMITOR, ...formattedStatements].join(LINE_DELIMITOR);
+    const formattedOutput = [lintOverridesBuffer, headerBuffer, LINE_DELIMITOR, this.typeDefs, LINE_DELIMITOR, ...formattedStatements].join(
+      LINE_DELIMITOR,
+    );
     return formattedOutput;
   }
 
+  buildTypeTag(name) {
+    if (!this.opTypeName) return '';
+    if (this.language !== 'typescript') return '';
+
+    const titleCasedName = `${name[0].toUpperCase()}${name.slice(1)}`;
```

Comment:
If I'm reading things right, what I'm deriving here is actually `name` from the `ops` arrays earlier in the chain. It *looks* safe to use this name and just fix casing, but in case I'm wrong and to prevent drift, I'm looking at adding an optional flag to `generateGraphQLDocuments` to return something like `Map<string, {query: string, meta: { ...}}>` when needed.

Reply:
Added as another PR to compare/contrast: https://github.com/aws-amplify/amplify-codegen/pull/623/files